src/utils/auth/auth.utils.ts

```diff
@@ -207,39 +207,47 @@ export async function createAuthPopup(authRequestData: null | AuthRequestData, m
   }
 }
 
+let isListeningForAuthResult = false;
+
 type AuthResultCallback<T> = (data: T) => void;
 
 const authResultCallbacks = new Map<string, AuthResultCallback<any>>();
 
-function addAuthResultListener<T>(authID: string, popupWindowTabID: number, fn: AuthResultCallback<T>) {
+function addAuthResultListener<T>(authID: string, fn: AuthResultCallback<T>) {
   authResultCallbacks.set(authID, fn);
 
-  if (authResultCallbacks.size === 1) {
-    isomorphicOnMessage("auth_result", ({ sender, data }) => {
```

Comment:
This would call `isomorphicOnMessage` more than once, as `authResultCallbacks` is constantly growing and shrinking, so `authResultCallbacks.size === 1` will be `true` multiple times.